src/routes/cache-hooks/cache-hooks.service.ts

```diff
@@ -28,7 +33,30 @@ export class CacheHooksService {
     private readonly safeRepository: ISafeRepository,
     @Inject(LoggingService)
     private readonly loggingService: ILoggingService,
-  ) {}
+    @Inject(IQueueConsumerService)
+    private readonly queueConsumerService: IQueueConsumerService,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+  ) {
+    this.queue = this.configurationService.getOrThrow<string>('amqp.queue');
+  }
+
+  onModuleInit(): Promise<void> {
+    return this.queueConsumerService.subscribe(
+      this.queue,
+      async (msg: ConsumeMessage) => {
+        try {
+          const content = JSON.parse(msg.content.toString());
+          const event: Event = WebHookSchema.parse(
+            JSON.parse(Buffer.from(content.data).toString()),
+          );
+          await this.onEvent(event);
+        } catch (err) {
+          this.loggingService.error(err);
+        }
+      },
+    );
+  }
```

Comment:
Here I made `CacheHooksService` implement `OnModuleInit` and take responsibility for subscribing to the queue. I'd prefer to have this kind of functionality on the `domain` layer, but it needs the `onEvent` function. So I think there are two options for a future improvement:
- Moving the `onEvent` function (along with all events entities) to the `domain` (by creating an `EventsRepository`).
- Subscribe from the `domain` (by creating an `EventsRepository`) and call this `onEvent` function in the `routes` layer.

I'm leaning towards the first option. Thoughts?